integration_tests/treasury_test.go

```diff
@@ -305,6 +318,248 @@ func TestTreasuryContract(t *testing.T) {
+	newAdminUser, err := ibctest.GetAndFundTestUserWithMnemonic(ctx, "newadmin", "", fundAmount, xion)
+	require.NoError(t, err)
+
+	// Test ProposeAdmin
+	t.Log("Testing ProposeAdmin")
+	proposeAdminMsg := map[string]interface{}{
+		"propose_admin": map[string]interface{}{
+			"new_admin": newAdminUser.FormattedAddress(),
+		},
+	}
+	proposeAdminMsgBz, err := json.Marshal(proposeAdminMsg)
+	require.NoError(t, err)
+
+	_, err = executeTreasuryMsg(t, ctx, xion, xionUser, treasuryAddr, proposeAdminMsgBz)
+	require.NoError(t, err)
+
+	// Query PendingAdmin
+	pendingAdminQuery := map[string]interface{}{"pending_admin": map[string]interface{}{}}
+	var pendingAdminRes map[string]interface{}
+	err = xion.QueryContract(ctx, treasuryAddr, pendingAdminQuery, &pendingAdminRes)
+	require.NoError(t, err)
+
+	// Test CancelProposedAdmin
+	t.Log("Testing CancelProposedAdmin")
+	cancelAdminMsg := map[string]interface{}{
+		"cancel_proposed_admin": map[string]interface{}{},
+	}
+	cancelAdminMsgBz, err := json.Marshal(cancelAdminMsg)
+	require.NoError(t, err)
+
+	_, err = executeTreasuryMsg(t, ctx, xion, xionUser, treasuryAddr, cancelAdminMsgBz)
+	require.NoError(t, err)
+
+	// Test UpdateGrantConfig
+	t.Log("Testing UpdateGrantConfig")
+	testAuth2 := authz.NewGenericAuthorization("/" + proto.MessageName(&banktypes.MsgMultiSend{}))
+	testGrant2, err := authz.NewGrant(time.Now(), testAuth2, &inFive)
+	require.NoError(t, err)
+
+	updatedGrantConfig := GrantConfig{
+		Description: "updated authorization",
+		Authorization: ExplicitAny{
+			TypeURL: testGrant2.Authorization.TypeUrl,
+			Value:   testGrant2.Authorization.Value,
+		},
+		Optional: true,
+	}
+
+	updateGrantConfigMsg := map[string]interface{}{
+		"update_grant_config": map[string]interface{}{
+			"msg_type_url": testAuth.MsgTypeURL(),
+			"grant_config": updatedGrantConfig,
+		},
+	}
+	updateGrantConfigMsgBz, err := json.Marshal(updateGrantConfigMsg)
+	require.NoError(t, err)
+
+	_, err = executeTreasuryMsg(t, ctx, xion, xionUser, treasuryAddr, updateGrantConfigMsgBz)
+	require.NoError(t, err)
+
+	// Test UpdateFeeConfig
+	t.Log("Testing UpdateFeeConfig")
+	newAllowance := feegrant.BasicAllowance{
+		SpendLimit: types.NewCoins(types.NewCoin("uxion", math.NewInt(1000))),
+		Expiration: &inFive,
+	}
+
+	newFeeGrant, err := feegrant.NewGrant(xionUserAddr, xionUserAddr, &newAllowance)
+	require.NoError(t, err)
+	updateFeeConfigMsg := map[string]interface{}{
+		"update_fee_config": map[string]interface{}{
+			"fee_config": FeeConfig{
+				Description: "updated fee grant",
+				Allowance: &ExplicitAny{
+					TypeURL: newFeeGrant.Allowance.TypeUrl,
+					Value:   newFeeGrant.Allowance.Value,
+				},
+				Expiration: int32(36000),
+			},
+		},
+	}
+	updateFeeConfigMsgBz, err := json.Marshal(updateFeeConfigMsg)
+	require.NoError(t, err)
+
+	_, err = executeTreasuryMsg(t, ctx, xion, xionUser, treasuryAddr, updateFeeConfigMsgBz)
+	require.NoError(t, err)
+
+	// Test UpdateParams
```

Comment:
there is also a update_treasury_params test that tests this also. we can try and consolidate them otherwise this can be removed